client/post-editor/editor-gutenberg-blocks-warning-dialog/index.jsx

```diff
@@ -0,0 +1,143 @@
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getEditorRawContent, getEditorPostId } from 'state/ui/editor/selectors';
+import Dialog from 'components/dialog';
+import { setSelectedEditor } from 'state/selected-editor/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditedPostValue } from 'state/posts/selectors';
+import getGutenbergEditorUrl from 'state/selectors/get-gutenberg-editor-url';
+import { openPostRevisionsDialog } from 'state/posts/revisions/actions';
+
+class EditorGutenbergBlocksWarningDialog extends Component {
+	static propTypes = {
+		translate: PropTypes.func,
+		postContent: PropTypes.string,
+		siteId: PropTypes.number,
+		gutenbergUrl: PropTypes.string,
+		switchToGutenberg: PropTypes.func,
+		openPostRevisionsDialog: PropTypes.func,
+	};
+
+	static defaultProps = {
+		translate: identity,
+		postContent: null,
+		siteId: null,
+		gutenbergUrl: null,
+		switchToGutenberg: noop,
+		openPostRevisionsDialog: noop,
+	};
+
+	state = {
+		isDialogVisible: false,
+		forceClassic: false,
+	};
+
+	shouldComponentUpdate( nextProps, nextState ) {
+		return this.state.isDialogVisible !== nextState.isDialogVisible;
+	}
+
+	static getDerivedStateFromProps( props, state ) {
+		const { postContent } = props;
+		const { forceClassic } = state;
+
+		const hasGutenbergBlocks = content => !! content && content.indexOf( '<!-- wp:' ) !== -1;
+
+		const isDialogVisible = ! forceClassic && hasGutenbergBlocks( postContent );
+
+		return {
+			isDialogVisible,
+		};
+	}
+
+	onUseClassic = () => {
+		this.setState( {
+			forceClassic: true,
+		} );
+	};
+
+	onSwitchToGutenberg = () => {
+		const { switchToGutenberg, siteId, gutenbergUrl } = this.props;
+		switchToGutenberg( siteId, gutenbergUrl );
+	};
+
+	onShowDocumentHistory = e => {
+		e.preventDefault();
+		this.props.openPostRevisionsDialog();
+		this.onUseClassic();
+	};
+
+	render() {
+		const { translate } = this.props;
+		const { isDialogVisible } = this.state;
+		const buttons = [
+			{
+				action: 'gutenberg',
+				label: translate( 'Switch to the new editor' ),
+				onClick: this.onSwitchToGutenberg,
+				isPrimary: true,
+			},
+			{
+				action: 'cancel',
+				label: translate( 'Use the classic editor' ),
+				onClick: this.onUseClassic,
+			},
+		];
+		return (
+			<Dialog isVisible={ isDialogVisible } buttons={ buttons } onClose={ this.onUseClassic }>
+				<h1>{ translate( 'This post uses blocks from the new editor' ) }</h1>
+
+				<p>
+					{ translate(
+						'You can continue to edit this post in the Classic Editor, but you may lose some data and formatting.'
+					) }
+				</p>
+
+				<p>
+					{ translate(
+						'You can also check the {{a}}document history{{/a}} and restore a version of the page from earlier.',
+						{
+							components: {
+								//eslint-disable-next-line jsx-a11y/anchor-is-valid
+								a: <a href="#" onClick={ e => this.onShowDocumentHistory( e ) } />,
```

Comment:
`e` and the arrow-function isn't necessary…I think we can pass `{ this.showDocumentHistory }` instead